Tests/Service/Request/RequestFactoryTest.php

```diff
@@ -166,12 +166,14 @@ public function testBuildFlow()
     public function testBuildFlowWithQueryParams()
     {
         $baseUrl = 'baseUrl';
-        $routeString = '/routeString?first-param={firstParam}&second-param=ignored value';
+        $routeString = '/routeString?first-param={firstParam}&second-param=ignored value&arr-param[key]={arrParam}&dot.seprate[0].array={dotParam}';
         $originParamValue = 'value with whitespaces';
+        $originArrParamValue = 'arrValue';
+        $originDotParamValue = 'dotValue';
         $requestMethod = 'GET';
         $requestBody = '';
 
-        $expectedUri = 'baseUrl/routeString?first-param=value+with+whitespaces&second-param=ignored value';
+        $expectedUri = 'baseUrl/routeString?first-param=value+with+whitespaces&second-param=ignored value&arr-param[key]=arrValue&dot.seprate[0].array=dotValue';
```

Comment:
I think, we face with array representing again.
How it will work with `&arr-param[key1]=arrValue1&arr-param[key2]=arrayValue2`?

Reply:
Why should we care? It never worked with array parameters before and now it will work with specific notations we need to support.

Reply:
But this change does not modify how the parameters are being parse since that is handle by the native way. This change allows to get the values which are the ones expected to be parsed.
This parsing is not use to generate the path, is just to extract the variable names. 
Since the parse_str returns can be a collection of string or array, and the array can be an string or array given the complexity of the query this is to just flatten it and get the values that is what is use for the replacement on the path.

Reply:
Since this is a public package, there are could be "valid workaround" that we could break with current changes. Meanwhile, we do not provide any "unified" solution with these changes.
Also, we create "specific notation" that already contain "bug" and unclear behavior.

Reply:
> "valid workaround"

This hypothetical work-around would be a usage of undocumented behavior. We cannot support undocumented behavior simply because we don't know about it and no one can possibly expect it from us.

> Meanwhile, we do not provide any "unified" solution with these changes.

There's no standard to provide a unified solution. But more importantly it's not the goal to provide one.

> we create "specific notation" 

We do not create it we are adding support for it. There might be potentially unlimited number of notations and we can only support limited number of them.

> that already contain "bug" and unclear behavior.

Can you point to the bug so we could fix it?